catalog/spatial/geocoding/spatial-geocoding-websearch/src/main/java/org/codice/ddf/spatial/geocoder/geonames/GeoNamesWebService.java

```diff
@@ -71,111 +73,80 @@
     private static final String PLACENAME_KEY = "name";
 
     @Override
-    public GeoResult getLocation(String location) {
+    public List<GeoEntry> query(String queryString, int maxResults) throws GeoEntryQueryException {
 
-        location = getUrlEncodedLocation(location);
+        List<GeoEntry> entries = new ArrayList<>();
+        String location = getUrlEncodedLocation(queryString);
 
-        String urlStr = String.format("%s://%s/searchJSON?q=%s&username=%s",
+        String urlStr = String.format("%s://%s/searchJSON?q=%s&username=%s&maxRows=%d",
```

Comment:
For the GeoNames Web service, the max allowed value for `maxRows` is 1000: http://www.geonames.org/export/geonames-search.html.

If `maxRows` exceeds that, an error message will be returned in your JSON: http://api.geonames.org/searchJSON?q=london&maxRows=10000&username=demo.

As-is, the code will just return an empty list in that case. I think we could either set `maxRows` to 1000 if the client passes a number greater than 1000 or throw an exception.